category.go

```diff
@@ -1,49 +1,56 @@
 package cli
 
-// CommandCategories is a slice of *CommandCategory.
+// CommandCategories wraps a slice of *CommandCategory.
 type CommandCategories struct {
-	Categories []*CommandCategory
+	slice []*CommandCategory
 }
 
 func NewCommandCategories() *CommandCategories {
-	return &CommandCategories{Categories: []*CommandCategory{}}
-}
-
-// CommandCategory is a category containing commands.
-type CommandCategory struct {
-	Name     string
-	Commands []*Command
+	return &CommandCategories{slice: []*CommandCategory{}}
 }
 
 func (c *CommandCategories) Less(i, j int) bool {
-	return c.Categories[i].Name < c.Categories[j].Name
+	return c.slice[i].Name < c.slice[j].Name
 }
 
 func (c *CommandCategories) Len() int {
-	return len(c.Categories)
+	return len(c.slice)
 }
 
 func (c *CommandCategories) Swap(i, j int) {
-	c.Categories[i], c.Categories[j] = c.Categories[j], c.Categories[i]
+	c.slice[i], c.slice[j] = c.slice[j], c.slice[i]
 }
 
-// AddCommand adds a command to a category.
-func (c *CommandCategories) AddCommand(category string, command *Command) *CommandCategories {
-	for _, commandCategory := range c.Categories {
+// AddCommand adds a command to a category, creating a new category if necessary.
+func (c *CommandCategories) AddCommand(category string, command *Command) {
```

Comment:
:heart:

On Mon, May 23, 2016, 22:55 Jesse Szwedko <notifications@github.com> wrote:

> In category.go
> <https://github.com/urfave/cli/pull/421#discussion_r64321853>:
>
> >  }
> >
> > -// AddCommand adds a command to a category.
> > -func (c *CommandCategories) AddCommand(category string, command *Command) *CommandCategories {
> > -	for _, commandCategory := range c.Categories {
> > +// AddCommand adds a command to a category, creating a new category if necessary.
> > +func (c *CommandCategories) AddCommand(category string, command *Command) {
>
> Missed that this is the existing interface :) Ignore me.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/urfave/cli/pull/421/files/cd10b49473a6178b5225df751c143cf07359439e..c70993dbc5080c379e8dc117d15a5f1f8e2775ca#r64321853>
>